### DISCUSSION TITLE
Fuzzing MQTT broker

### DISCUSSION THREAD

**Discussion by giacomochech** [jtpereyda/boofuzz#700] [Q&A]
Are there any pre-made examples of a script for fuzzing an MQTT broker? I am new to the tool and I am having a bit of a hard time creating the script from scratch.
In case it doesn't exist how can I approach the documentation (which parts are most useful and which parts I may not learn)? Are there any suggestion to start using BooFuzz?
Thank you very much!

**Comment by SR4ven** [comment]
Currently we don't have any examples for MQTT.

But you can take a look at other protocol [definitions](https://github.com/jtpereyda/boofuzz/tree/master/request_definitions) and [examples](https://github.com/jtpereyda/boofuzz/tree/master/examples).
Then see how the protocol you're trying to fuzz is built, docs, Wireshark, ... and use the blocks and primitives of boofuzz to rebuild it. Those are more or less well documented.

I hope that helps. If you have any more specific questions feel free to ask.
If you get some MQTT example running, don't hesitate to contriute it!
